File: Chatterly_Android/app/src/main/java/com/example/chatterly/model/data/Message.java
package com.example.chatterly.model.data;

import java.util.Date;
import java.util.List;
import java.util.UUID;

public class Message {
    private final int id;
    private String uid;
    private final String body;
    private final List<String> images;
    private final List<String> videos;
    private final List<String> audios;
    private final Date publishDate;
    private Integer messageRepliedTo;
    private final String authorId;
    private int reactionsCount;
    private int repliesCount;
    private final int chatId;
    private Chat chat;
    private User author;

    public Message(int id, String body, List<String> images, List<String> videos, List<String> audios,
                   Date publishDate, String authorId, int reactionsCount, int repliesCount, int chatId) {
        this.id = id;
        this.body = body;
        this.images = images;
        this.videos = videos;
        this.audios = audios;
        this.publishDate = publishDate;
        this.authorId = authorId;
        this.reactionsCount = reactionsCount;
        this.repliesCount = repliesCount;
        this.chatId = chatId;
        this.uid = UUID.randomUUID().toString();
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User author) {
        this.author = author;
    }

    public Chat getChat() {
        return chat;
    }

    public void setChat(Chat chat) {
        this.chat = chat;
    }

    public int getRepliesCount() {
        return repliesCount;
    }

    public void setRepliesCount(int repliesCount) {
        this.repliesCount = repliesCount;
    }

    public int getReactionsCount() {
        return reactionsCount;
    }

    public void setReactionsCount(int reactionsCount) {
        this.reactionsCount = reactionsCount;
    }

    public Integer getMessageRepliedTo() {
        return messageRepliedTo;
    }

    public void setMessageRepliedTo(Integer messageRepliedTo) {
        this.messageRepliedTo = messageRepliedTo;
    }

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public int getChatId() {
        return chatId;
    }

    public String getAuthorId() {
        return authorId;
    }

    public Date getPublishDate() {
        return publishDate;
    }

    public List<String> getAudios() {
        return audios;
    }

    public List<String> getVideos() {
        return videos;
    }

    public List<String> getImages() {
        return images;
    }

    public String getBody() {
        return body;
    }

    public int getId() {
        return id;
    }
}

File: Chatterly_Backend/Data/Models/Message.cs
﻿using Chatterly_Backend.Data.DTOs;

namespace Chatterly_Backend.Data.Models
{
    public class Message
    {
        public required int Id { get; set; }
        public required string Uid { get; set; }
        public required string Body { get; set; }
        public List<string> Images { get; set; } = [];
        public List<string> Videos { get; set; } = [];
        public List<string> Audios { get; set; } = [];
        public DateTime PublishDate { get; set; } = DateTime.Now;
        public int? MessageRepliedTo { get; set; }
        public required string AuthorId { get; set; }
        public int ReactionsCount { get; set; }
        public int RepliesCount { get; set; }
        public required int ChatId { get; set; }

        public virtual User? Author { get; set; }
        public virtual Chat? Chat { get; set; }
        public virtual ICollection<MessageReaction> Reactions { get; set; } = [];
        public virtual ICollection<User> Mentions { get; set; } = [];

        public MessageDTO ToDTO()
        {
            return new MessageDTO
            {
                Id = Id,
                Uid = Uid,
                Body = Body,
                Images = Images,
                Videos = Videos,
                Audios = Audios,
                PublishDate = PublishDate,
                MessageRepliedTo = MessageRepliedTo,
                AuthorId = AuthorId,
                ReactionsCount = ReactionsCount,
                RepliesCount = RepliesCount,
                ChatId = ChatId,
            };
        }
    }
}
